definitions/procedures/pulp/remove.rb

```diff
@@ -163,12 +174,33 @@ def drop_migrations
     end
     # rubocop:enable Metrics/BlockLength
 
-    def delete_pulp_data(rm_cmds)
+    def delete_pulp_data
+      non_mountpoints = data_dir_removal_cmds(deletable_pulp_dirs)
+      mountpoints = pulp_data_dirs_mountpoints
       with_spinner('Deleting pulp2 data directories') do |spinner|
-        rm_cmds.each do |cmd|
-          execute!(cmd)
+        if non_mountpoints.any?
```

Comment:
I'd put the `any?` outside the `with_spinner` block, as otherwise you'd possibly end up with displaying "Deleting pulp2 data" without doing anything if *everything* is a mointpoint (I don't think those setups exist, but we still can be correct ;))

Reply:
done!

Reply:
> (I don't think those setups exist, but we still can be correct ;))

One possible scenario is that no directories exist and only a mountpoint remains. Still unlikely but a lot more likely than the scenario where there are only mountpoints :)

Reply:
Btw, I think I still agree with @evgeni that it should be (roughly)
```ruby
if non_mountpoints.any?
  with_spinner('') do |spinner|
    # ...
  end
end
```